controller/src/beerocks/master/db/db.cpp

```diff
@@ -3190,6 +3190,8 @@ std::list<db::bss_info_conf_t> &db::get_bss_info_configuration(const sMacAddr &a
 
 void db::clear_bss_info_configuration() { bss_infos.clear(); }
 
+void db::clear_bss_info_configuration(const sMacAddr &al_mac) { bss_infos[al_mac].clear(); }
```

Comment:
It will be more suitable to edit the exist clear function and add it to it an optional value of MacAddr.
Of course that it will cause the function to be with more that one line so I will be needed to move the implementation to the CPP file.

Reply:
> It will be more suitable to edit the exist clear function and add it to it an optional value of MacAddr.

I disagree, I think it makes more sense that it's an overload. An optional argument has two disadvantages:

- It needs to be a pointer (otherwise it's not optional) which makes the code slightly more difficult to read.
- The implementation will simply have a condition on the argument being nullptr or not; two different function signature are much easier to read.

> Of course that it will cause the function to be with more that one line so I will be needed to move the implementation to the CPP file.

It already is in the cpp file :-)

Reply:
> It already is in the cpp file :-)

Yeah I confused 😏 

> >It will be more suitable to edit the exist clear function and add it to it an optional value of MacAddr.
> 
> I disagree, I think it makes more sense that it's an overload. An optional argument has two disadvantages:
> 
> It needs to be a pointer (otherwise it's not optional) which makes the code slightly more difficult to read.
> The implementation will simply have a condition on the argument being nullptr or not; two different function signature are much easier to read.

It does not have to be a pointer, it can still be a reference. I tested the following code and it worked:

```cpp
#include <iostream>
#include <string>

std::string ret_str(const std::string &str = "hello world") 
{
  return str;
}

int main() {
  std::cout << ret_str() << std::endl;
}
```
`output: hello world`

Reply:
> > > It will be more suitable to edit the exist clear function and add it to it an optional value of MacAddr.
> > I disagree, I think it makes more sense that it's an overload. An optional argument has two disadvantages:
> > It needs to be a pointer (otherwise it's not optional) which makes the code slightly more difficult to read.
> It does not have to be a pointer, it can still be a reference. I tested the following code and it worked:
> ```c++
> std::string ret_str(const std::string &str = "hello world") 
> ```

Yes, but what would you give as default?

```
    void clear_bss_info_configuration(const sMacAddr &al_mac = ????);
```

You could use `ZERO_MAC`, indeed, then the issue with the pointer is gone. But still you would need a condition in the implementation, and you'd need it as well in the documentation (which is now missing anyway). If you have it as separate function signatures, the condition is implied.

Clearly this is a nitpicking issue, but it would be good to decide if we in general prefer optional arguments to be declared with default parameter values and a condition in the implementation, or as two separate overloaded functions (where one potentially calls the other).